app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/armor/detail/ArmorSetDetailFragment.kt

```diff
@@ -0,0 +1,198 @@
+        armor_set_defense_value.text = getString(R.string.armor_defense_value,
+                armorSet.armor.sumBy { armorFull -> armorFull.armor.defense_base },
+                armorSet.armor.sumBy { armorFull -> armorFull.armor.defense_max },
+                armorSet.armor.sumBy { armorFull -> armorFull.armor.defense_augment_max })
+
+        //Calculate total materials
+        populateSetComponents(armorSet.armor)
+    }
+
+    private fun populateArmorSetPieces(armorFull: ArmorFull) {
+        val view = layoutInflater.inflate(R.layout.listitem_armorset_armor, armor_set_piece_list, false)
+
+        view.armor_icon.setImageDrawable(AssetLoader.loadIconFor(armorFull.armor))
+        view.armor_name.text = armorFull.armor.name
+        view.rarity_string.text = getString(R.string.format_rarity, armorFull.armor.rarity)
+        view.defense_value.text = view.resources.getString(
+                R.string.armor_defense_value,
+                armorFull.armor.defense_base,
+                armorFull.armor.defense_max,
+                armorFull.armor.defense_augment_max)
+        val slotImages = armorFull.armor.slots.map {
+            view.context.getDrawableCompat(SlotEmptyRegistry(it))
+        }
+
+        view.slot1.setImageDrawable(slotImages[0])
+        view.slot2.setImageDrawable(slotImages[1])
+        view.slot3.setImageDrawable(slotImages[2])
+
+        view.setOnClickListener {
+            getRouter().navigateArmorDetail(armorFull.armor.id)
+        }
+
+        armor_set_piece_list.addView(view)
+    }
+
+    private fun populateSetBonuses(armorSetBonuses: List<ArmorSetBonus>) {
+        if (armorSetBonuses.isEmpty()) {
+            armor_set_set_bonus_section.visibility = View.GONE
+            return
+        }
+
+        // show set bonus section
+        armor_set_set_bonus_section.visibility = View.VISIBLE
+        armor_set_set_bonus_list.removeAllViews()
+
+        //Set the label for the Set name
+        armor_set_set_bonus_name.text = armorSetBonuses.first().name
+
+        //Now to set the actual skills
+        for (setBonus in armorSetBonuses) {
+            val skillIcon = AssetLoader.loadIconFor(setBonus.skillTree)
+            val reqIcon = SetBonusNumberRegistry(setBonus.required)
+            val listItem = layoutInflater.inflate(R.layout.listitem_armorset_bonus, null)
+
+            listItem.bonus_skill_icon.setImageDrawable(skillIcon)
+            listItem.bonus_skill_name.text = setBonus.skillTree.name
+            listItem.bonus_requirement.setImageResource(reqIcon)
+
+            listItem.setOnClickListener {
+                getRouter().navigateSkillDetail(setBonus.skillTree.id)
+            }
+
+            armor_set_set_bonus_list.addView(listItem)
+        }
+    }
+
+    private fun populateSetComponents(armorSet: List<ArmorFull>) {
+        val components = mutableMapOf<Int, ItemQuantity>()
+        armor_set_components_list.removeAllViews()
+
+        armorSet.forEach { armorFull ->
```

Comment:
You might want a comment up here because it isn't immediately obvious how this works.